sdk/core/ci.yml

```diff
@@ -42,6 +42,55 @@ stages:
       LiveTestTimeoutInMinutes: 90 # default is 60 min. We need a little longer on worst case for Win+jsonTests
       LineCoverageTarget: 93
       BranchCoverageTarget: 55
+      PreTestSteps:
+        - task: UsePythonVersion@0
+          displayName: 'Use Python 3'
+          inputs:
+            versionSpec: '3'
+          condition: and(succeeded(), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'))
+
+        - pwsh: |
+            python --version
+            pip install -r requirements.txt
+          workingDirectory: build/sdk/core/azure-core/test/ut
+          displayName: Install Python requirements.
+          condition: and(succeeded(), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'))
+        - pwsh: |
+            Start-Process 'python.exe' `
+                -ArgumentList 'sdk/core/azure-core/test/ut/websocket_server.py' `
+                -NoNewWindow -PassThru -RedirectStandardOutput $(Build.SourcesDirectory)/WebSocketServer.log
+          workingDirectory: build
+          condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'))
+          displayName: Launch python websocket server (Windows).
+
+        - bash: |
+            nohup python sdk/core/azure-core/test/ut/websocket_server.py > $(Build.SourcesDirectory)/WebSocketServer.log &
+          workingDirectory: build
+          condition: and(succeeded(), ne(variables['Agent.OS'], 'Windows_NT'), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'))
+          displayName: Launch python websocket server (Linux).
+
```

Comment:
We can wrap this up into a single PowerShell file (PowerShell is installed on all the agent OS') 

For example: `build/sdk/core/azure-core/test/ut/Start-WebSocketServer.ps1` (assuming you put `Start-WebSocketServer.ps1` in the same folder as `websocket_server.py` and the build copies the file over) 

```pwsh
param(
  [string] $LogFileLocation = "$($env:BUILD_SOURCESDIRECTORY)/WebSocketServer.log" 
)

if ($IsWindows) { 
  Start-Process 'python.exe' `
    -ArgumentList 'websocket_server.py' `
    -NoNewWindow -PassThru -RedirectStandardOutput $(Build.SourcesDirectory)/WebSocketServer.log
} else { 
  nohup python websocket_server.py > $(Build.SourcesDirectory)/WebSocketServer.log &
}
```

Now you can call it once and don't have to mostly duplicate tasks and differentiate by conditions `ne(variables['Agent.OS'], 'Windows_NT')` 

```yaml
- PowerShell@2
  displayName: Launch Python WebSocket Server
  inputs: 
    pwsh: true
    filePath: Start-WebSocketServer.ps1
    workingDirectory: build/sdk/core/azure-core/test/ut/
```